packages/webpack/lib/middleware/agent.js

```diff
@@ -0,0 +1,18 @@
+'use strict';
+
+const { matchesUA } = require('browserslist-useragent');
+
+module.exports = function createAgentMiddleware({ config: { browsers } }) {
```

Comment:
I really like the idea and the feature but wonder if it makes more sense in hops as this seems a bit too far from the core tooling platform untool ought to be.

Also, a test would be highly appreciated.